tools/lib/logreader.py

```diff
@@ -92,18 +92,32 @@ def default_valid_file(fn: LogPath) -> bool:
 
 
 def auto_strategy(rlog_paths: LogPaths, qlog_paths: LogPaths, interactive: bool, valid_file: ValidFileCallable) -> LogPaths:
-  # auto select logs based on availability
-  if any(rlog is None or not valid_file(rlog) for rlog in rlog_paths) and all(qlog is not None and valid_file(qlog) for qlog in qlog_paths):
-    if interactive:
-      if input("Some rlogs were not found, would you like to fallback to qlogs for those segments? (y/n) ").lower() != "y":
-        return rlog_paths
-    else:
-      cloudlog.warning("Some rlogs were not found, falling back to qlogs for those segments...")
-
-    return [rlog if valid_file(rlog) else (qlog if valid_file(qlog) else None)
-            for (rlog, qlog) in zip(rlog_paths, qlog_paths, strict=True)]
+  # Count missing files
+  missing_files_count = count_missing_files(rlog_paths, qlog_paths, valid_file)
+
+  if missing_files_count != 0:
+    cloudlog.error(f"Error: {missing_files_count} segments are missing either rlogs or qlogs.")
+    # auto select logs based on availability
+    if any(rlog is None or not valid_file(rlog) for rlog in rlog_paths) and all(qlog is not None and valid_file(qlog) for qlog in qlog_paths):
```

Comment:
can we remove this line if we check `missing_files_count` right above?

Reply:
No, because there could be scenario with qlogs missing.